schema/gnosis_safe/view_safes.sql

```diff
@@ -0,0 +1,66 @@
+            --     When Gnosis relayer triggered ProxyFactory via createProxyWithNonce
+            --         and 3 or 4 owners
+            --         and threshold 1 or 2 then Personal Safe
+            --     When ProxyFactory was triggered via createProxy by something other than the Gnosis relayer then we consider it a Team Safe
+            --     When something else, then unknown (e.g. no factory used or createProxyWithNonce by something other than relay service
+            --        If unknown is increasing, we could drill down more.
+            case
+                when
+                    (owners = 3 or owners = 4)
+                    and (threshold = 1 or threshold = 2)
+                    and (
+                        (et.to = '\x8942595A2dC5181Df0465AF0D7be08c8f23C93af'  -- master copy address v0.1.0
+                            and (et2.to is null or et2.to <> '\x88cd603a5dc47857d02865bbc7941b588c533263')
+                        )
+                        or (et.to in ('\xb6029ea3b2c51d09a50b53ca8012feeb05bda35a', '\xae32496491b53841efb51829d6f886387708f99b', '\x34cfac646f301356faa8b21e94227e3583fe3f5f') -- mastercopy address v1.0.0, v1.1.0, v1.1.0, v1.1.1
+                            and et2.from = '\x07f455f30e862e13e3e3d960762cb11c4f744d52'  -- relay service
+                            and et2.to in ('\x12302fe9c02ff50939baaaaf415fc226c078613c', '\x76e2cfc1f5fa8f6a5b3fc4c8f4788f0116861f9b')  -- ProxyFactory v1.0.0, v1.1.1
+                            and left(text(et2."input"),10) = '\x1688f0b9' -- createProxyWithNonce method
+                        )
+                    )
+                then 'personal'
+                when
+                    et.to in ('\xb6029ea3b2c51d09a50b53ca8012feeb05bda35a', '\xae32496491b53841efb51829d6f886387708f99b', '\x34cfac646f301356faa8b21e94227e3583fe3f5f') -- mastercopy address v1.0.0, v1.1.0, v1.1.1
+                    and et2.from <> '\x07f455f30e862e13e3e3d960762cb11c4f744d52'  -- relay service
+                    and et2.to in ('\x12302fe9c02ff50939baaaaf415fc226c078613c', '\x76e2cfc1f5fa8f6a5b3fc4c8f4788f0116861f9b')  -- ProxyFactory v1.0.0, v1.1.1
+                    and left(text(et2."input"),10) in ('\x61b69abd', '\x1688f0') -- createProxy method
+                then 'team'
+                else 'unknown'
+                end
+             as safe_type
+        from ethereum.traces et -- for the setup calls
+        left outer join safe_setups ss
+            on et.tx_hash = ss.tx_hash
+        left outer join ethereum.traces et2  -- for the potential proxy calls
+            on et.tx_hash = et2.tx_hash and et2.call_type = 'call' and left(text(et2."input"),10) in ('\x1688f0b9', '\x61b69abd') -- createProxyWithNonce or createProxy method call
+        where et.tx_success = True
+            AND left(text(et."input"),10) in ('\x0ec78d9e', '\xa97ab18a', '\xb63e800d') -- setup methods of v0.1.0, v1.0.0, v1.1.0=v.1.1.1
```

Comment:
I'm sorry I didn't see this on my first pass, but I think we can speed this up by using
`substring(et."input" for 4)` instead of `left(text(et."input"), 10)` throughout the query.

Reply:
No worries at all. If you think this makes it faster, I'm happy to change it :)
I also chanced it in the 2 other places inside the select statement of the "safes" subquery